torch/__init__.py

```diff
@@ -567,33 +567,35 @@ def _assert(condition, message):
 # Import most common subpackages
 ################################################################################
 
-import torch.cuda
-import torch.autograd
+# Use explicit redundant import to allow static type checkers to detect submodules
+
+from torch import cuda as cuda
+from torch import autograd as autograd
 from torch.autograd import no_grad, enable_grad, set_grad_enabled
-import torch.fft
-import torch.futures
-import torch.nn
+from torch import fft as fft
+from torch import futures as futures
+from torch import nn as nn
 import torch.nn.intrinsic
 import torch.nn.quantizable
 import torch.nn.quantized
-import torch.optim
+from torch import optim as optim
 import torch.optim._multi_tensor
-import torch.multiprocessing
-import torch.sparse
+from torch import multiprocessing as multiprocessing
+from torch import sparse as sparse
 import torch.utils.backcompat
-import torch.onnx
-import torch.jit
-import torch.linalg
-import torch.hub
-import torch.random
-import torch.distributions
-import torch.testing
+from torch import onnx as onnx
+from torch import jit as jit
+from torch import linalg as linalg
+from torch import hub as hub
+from torch import random as random
+from torch import distributions as distributions
+from torch import testing as testing
 import torch.backends.cuda
```

Comment:
Backends are also accessed via `import torch` and used like `torch.backends.cuda.is_built()` (https://pytorch.org/docs/stable/backends.html).  I guess this change as it is now won't fix those imports for pyright?

I care _more_ about the top-level imports but it'd be bummer if pyright would work only for a subset of imports.